master/internal/agent/agent.go

```diff
@@ -53,7 +53,12 @@ func (a *agent) Receive(ctx *actor.Context) error {
 		socket, ok := msg.Accept(ctx, aproto.MasterMessage{}, true)
 		check.Panic(check.True(ok, "failed to accept websocket connection"))
 		a.socket = socket
-		a.address = strings.SplitN(msg.Ctx.Request().RemoteAddr, ":", 2)[0]
+		lastColonIndex := strings.LastIndex(msg.Ctx.Request().RemoteAddr, ":")
+		if lastColonIndex == -1 {
+			a.address = msg.Ctx.Request().RemoteAddr
+		} else {
+			a.address = msg.Ctx.Request().RemoteAddr[0:lastColonIndex]
```

Comment:
Is the intention to grab everything except the port?

Reply:
Yes.  The idea here is to grab the address of the agent with the port so it can be used in opening a shell. The port of shell container is in a different field "host_port", retrieved from the docker daemon on the agents.

Reply:
Got it.